mutiny-core/src/lib.rs

```diff
@@ -2072,14 +2288,143 @@ pub(crate) async fn create_new_federation<S: MutinyStorage>(
         federation_name,
         federation_expiry_timestamp,
         welcome_message,
+        gateway_fees,
     })
 }
 
+// max amount that can be spent through a gateway
+fn max_spendable_amount(amount_sat: u64, routing_fees: GatewayFees) -> Option<u64> {
+    let amount_msat = amount_sat as f64 * 1_000.0;
+
+    let prop_fee_msat = (amount_msat * routing_fees.proportional_millionths as f64) / 1_000_000.0;
+
+    let initial_max = amount_msat - (routing_fees.base_msat as f64 + prop_fee_msat);
+
+    if initial_max <= 0.0 {
+        return None;
+    }
+
+    if amount_msat - initial_max < 1.0 {
+        return Some((initial_max / 1_000.0).floor() as u64);
+    }
+
+    let mut new_max = initial_max;
+    while new_max < amount_msat {
+        let new_check = new_max + 1.0;
+
+        let prop_fee_sat = (new_check * routing_fees.proportional_millionths as f64) / 1_000_000.0;
+
+        let new_amt = new_check + routing_fees.base_msat as f64 + prop_fee_sat;
```

Comment:
why is the fees added here?

Reply:
some comments in this function would help explain the logic of everything

Reply:
because there's a new amount, so it needs a new fee